oauthproxy.go

```diff
@@ -165,8 +165,13 @@ func NewWebSocketOrRestReverseProxy(u *url.URL, opts *Options, auth hmacauth.Hma
 		setProxyDirector(proxy)
 	}
 
-	wsURL := &url.URL{Scheme: "ws://", Host: u.Host}
-	wsProxy := wsutil.NewSingleHostReverseProxy(wsURL)
+	// this should give us a wss:// scheme if the url is https:// based.
+	var wsProxy *wsutil.ReverseProxy = nil
+	if opts.ProxyWebSockets {
+		wsScheme := "ws" + strings.TrimPrefix(u.Scheme, "http")
+		wsURL := &url.URL{Scheme: wsScheme, Host: u.Host}
+		wsProxy = wsutil.NewSingleHostReverseProxy(wsURL)
```

Comment:
I think you will also need to set up wsProxy.TLSClientConfig with the upstream CAs if using wss, e.g.,

		if wsScheme == "wss" && len(opts.UpstreamCAs) > 0 {
			pool, err := util.GetCertPool(opts.UpstreamCAs, false)
			if err != nil {
				log.Fatal("Failed to fetch CertPool: ", err)
			}
			wsProxy.TLSClientConfig = &tls.Config{
				RootCAs: pool,
			}
		}